.github/workflows/build-nightly.yaml

```diff
@@ -210,3 +210,112 @@ jobs:
           DATACORDER_USER: ${{ secrets.DATACORDER_USER }}
           DATACORDER_SSHPASS: ${{ secrets.DATACORDER_PASSWORD }}
         run: $GITHUB_WORKSPACE/ci/linux/upload_dist_package.sh nightly
+
+  build_mac:
+    strategy:
+      matrix:
+        configuration: [FastDebug, Release]
+        arch: [x86_64]
+    name: Mac
+    runs-on: macos-latest
+    steps:
+      - name: Cache Qt
+        id: cache-qt-lin
+        uses: actions/cache@v1
+        with:
+          path: ${{ github.workspace }}/../Qt
+          key: ${{ runner.os }}-QtCache-${{ env.QT_VERSION }}
+      - name: Install Qt
+        uses: jurplel/install-qt-action@v2
+        with:
+          version: ${{ env.QT_VERSION }}
+          dir: ${{ github.workspace }}/..
+          cached: ${{ steps.cache-qt-lin.outputs.cache-hit }}
+          setup-python: 'false'
+          aqtversion: ==1.1.3
+
+      - uses: actions/checkout@v1
+        name: Checkout
+        with:
+          submodules: true
+          fetch-depth: 0
+      - name: Set up test version
+        shell: bash
+        run: |
+          [[ "${{ github.ref }}" =~ ^refs\/heads\/test\/(.*)$ ]]
+          # Override the revision string so that the builds are named correctly
+          echo "set(FSO_VERSION_REVISION_STR ${BASH_REMATCH[1]})" > "version_override.cmake"
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          key: ${{ runner.os }}-${{ matrix.configuration }}-${{ matrix.compiler }}
+
+      - name: Configure CMake
+        env:
+          CONFIGURATION: ${{ matrix.configuration }}
+          COMPILER: ${{ matrix.compiler }}
+          ARCHITECTURE: ${{ matrix.arch }}
+          JOB_CMAKE_OPTIONS: ${{ matrix.cmake_options }}
+        run: |
+          mkdir build
+          cd build
+          
+          export CXXFLAGS="-mtune=generic -pipe -Wno-unknown-pragmas"
+          export CFLAGS="-mtune=generic -pipe -Wno-unknown-pragmas"
+          export CMAKE_OSX_ARCHITECTURES="$ARCHITECTURE"
+          
+          brew install ninja
+          cmake -G Ninja -DFSO_FATAL_WARNINGS=ON -DCMAKE_EXPORT_COMPILE_COMMANDS=ON $CMAKE_OPTIONS \
```

Comment:
Isn't appimage something that is Linux-specific though? I specifically excluded it because it was failing to build on Mac, and didn't seem like something we want to build on Mac...

Reply:
`-DCMAKE_C_COMPILER_LAUNCHER=ccache` is here to make use of `ccache` to speed up builds. The linux setup is a bit different, but it is apparently only activated if the compiler is `gcc`:
<img width="686" alt="image" src="https://user-images.githubusercontent.com/932048/164573705-c911af20-1aeb-45fe-8e05-1bbcb597b6e8.png">
Since I'm not sure why this is gated by `"${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU"`, to avoid changing this, I've chosen to pass `CMAKE_C_COMPILER_LAUNCHER` via command line.

Reply:
Ok, we'll see what CI thinks about it, then

Reply:
I tested it and it does cache and speed up subsequent builds (~13 minutes -> ~2 minutes).